climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,891 @@
+
+    return gdf_agg
+
+
+def poly_to_pnts(gdf, res):
+    """
+    Disaggragate (multi-)polygons geodataframe to points
+
+    Parameters
+    ----------
+    gdf : geodataframe
+        Can be any CRS
+    res : float
+        Resolution (same units as gdf crs)
+
+    Returns
+    -------
+    geodataframe
+        Geodataframe with a double index, first for polygon geometries,
+        second for the point disaggregation of the polygons.
+
+    """
+
+    # Needed because gdf.explode() requires numeric index
+    idx = gdf.index.to_list() #To restore the naming of the index
+
+    gdf_points = gdf.copy().reset_index(drop=True)
+    gdf_points['geometry_pnt'] = gdf_points.apply(
+        lambda row: _interp_one_poly(row.geometry, res), axis=1)
+    gdf_points = _swap_geom_cols(
+        gdf_points, geom_to='geometry_orig', new_geom='geometry_pnt'
+        )
+
+    gdf_points = gdf_points.explode()
+    gdf_points.index = gdf_points.index.set_levels(idx, level=0)
+    return gdf_points
+
+def poly_to_pnts_m(gdf, res):
+    """
+    Disaggragate (multi-)polygons geodataframe to points
+
+    Parameters
+    ----------
+    gdf : geodataframe
+        Can be any CRS
+    res : float
+        Resolution in meters
+
+    Returns
+    -------
+    geodataframe
+        Geodataframe with a double index, first for polygon geometries,
+        second for the point disaggregation of the polygons.
+
+    """
+
+    orig_crs = gdf.crs
+
+    # Needed because gdf.explode() requires numeric index
+    idx = gdf.index.to_list() # To restore the naming of the index.
+
+    gdf_points = gdf.copy().reset_index(drop=True)
+    gdf_points['geometry_pnt'] = gdf_points.apply(
+        lambda row: _interp_one_poly_m(row.geometry, res, orig_crs), axis=1)
+    gdf_points = _swap_geom_cols(
+        gdf_points, geom_to='geometry_orig', new_geom='geometry_pnt'
+        )
+    gdf_points = gdf_points.explode()
+    gdf_points.index = gdf_points.index.set_levels(idx, level=0)
+    return gdf_points
+
+def _interp_one_poly(poly, res):
+    """
+    Disaggragate a single polygon to points
+
+    Parameters
+    ----------
+    poly : shapely Polygon
+        Polygon
+    res : float
+        Resolution (same units as gdf crs)
+
+    Returns
+    -------
+    shapely multipoint
+        Grid of points rasterizing the polygon
+
+    """
+
+    if poly.is_empty:
+        return shgeom.MultiPoint([])
+
+    height, width, trafo = u_coord.pts_to_raster_meta(poly.bounds, (res, res))
+    x_grid, y_grid = u_coord.raster_to_meshgrid(trafo, width, height)
+    in_geom = sh.vectorized.contains(poly, x_grid, y_grid)
+
+    if sum(in_geom.flatten()) > 1:
+        return shgeom.MultiPoint([(x, y) for x, y in
+                            zip(x_grid[in_geom], y_grid[in_geom])])
+    else:
+        return shgeom.MultiPoint([poly.representative_point()])
+
+def _interp_one_poly_m(poly, res, orig_crs):
+    """
+    Disaggragate a single polygon to points. Resolution in meters.
```

Comment:
Explanation to how the crs needs to be adjusted to a geographically suitable one in metres.

Reply:
This is a private method, so I kept the comments to a minimum. But it could help future developers. What do you mean though? Here any crs can be used... The method does the projections itself.

Reply:
Yes yes. I mean the fact that the crs is projected in the code. Would be useful to other developers that are not geographers.